File: src/java/jnpp/service/dto/IdentityDTO.java
package jnpp.service.dto;

public class IdentityDTO {
    
    public enum Gender {
        
        MALE,
        FEMALE;
        
    }   
   
    private IdentityDTO.Gender gender;
    private String firstname;
    private String lastname;

    public IdentityDTO(IdentityDTO.Gender gender, String firstname, String lastname) {
        this.gender = gender;
        this.firstname = firstname;
        this.lastname = lastname;
    }

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }
        
    @Override
    public String toString() {
        return firstname + " " + lastname;
    }
    
}

File: src/main/webapp/commons/dto/identity-dto.js
class IdentityDTO {

    constructor(json) {
        if (!json || !json.gender || !json.firstname || !json.lastname)
            throw new ParseException(json, typeof this);
        this.gender = json.gender;
        this.street = json.street;
        this.city = json.city;
        this.state = json.state;
    }

}

IdentityDTO.Gender = Object.freeze({
    MALE: 0,
    FEMALE: 1
});
